File: OC图文混排复习/OC图文混排复习/ViewController.m
//
//  ViewController.m
//  OC图文混排复习
//
//  Created by jyh on 2017/12/15.
//  Copyright © 2017年 jyh. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *label;

@end

@implementation ViewController

- (void)viewDidLoad {
    
    [super viewDidLoad];
    
    // 附件
    NSTextAttachment *attachment = [[NSTextAttachment alloc] init];
    // 来源图片
    attachment.image = [UIImage imageNamed:@"lxh_haobang"];
    
    //lineHeight 大致和字体的大小相等
    CGFloat height = self.label.font.lineHeight;
    // 设置 大小
    attachment.bounds = CGRectMake(0, -4, height, height);
    
    // 图像字符串
    NSAttributedString *imageStr = [NSAttributedString attributedStringWithAttachment:attachment];

    // 定义一个可变的字符串
    NSMutableAttributedString *attrStrM = [[NSMutableAttributedString alloc] initWithString:@"我"];
    
    //拼接图片文本
    [attrStrM appendAttributedString:imageStr];
    
    [attrStrM appendAttributedString:[[NSMutableAttributedString alloc] initWithString:@"88!"]];
    
    // 设置属性文本
    self.label.attributedText = attrStrM;
    
    
}



@end

File: swift图文混排/swift图文混排/ViewController.swift
//
//  ViewController.swift
//  swift图文混排
//
//  Created by jyh on 2017/12/15.
//  Copyright © 2017年 jyh. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var label: UILabel!
    override func viewDidLoad() {
        super.viewDidLoad()
 
        // 图片附件
        let attachment = NSTextAttachment()
        
        // 图片
        attachment.image = #imageLiteral(resourceName: "lxh_haobang")
        
        let height = label.font.lineHeight
        attachment.bounds = CGRect(x: 0, y: -4, width: height, height: height)
        
        // 属性文本
        let imageStr = NSAttributedString(attachment: attachment)
        
        // 图文字符串
        let attrStrM = NSMutableAttributedString(string: "我")
        
        attrStrM.append(imageStr)
        attrStrM.append(NSMutableAttributedString(string: "88!"))
        
        
        label.attributedText = attrStrM
        
    }

}

